node/src/effect/requests.rs

```diff
@@ -829,6 +849,8 @@ impl<I: Display> Display for LinearChainRequest<I> {
 pub enum ConsensusRequest {
     /// Request for consensus to sign a new linear chain block and possibly start a new era.
     HandleLinearBlock(Box<BlockHeader>, Responder<Option<FinalitySignature>>),
+    /// Check whether validator identifying with the public key is bonded.
+    IsBondedValidator(EraId, PublicKey, Responder<bool>),
```

Comment:
I'd prefer if the auction contract kept track of the past few bonded eras as well. (But if it doesn't that's not for this PR, of course!) That would make this signature logic a bit less dependent on the consensus component.

Or is the idea that asking consensus is faster because it doesn't involve looking things up in the global state?

Reply:
> I'd prefer if the auction contract kept track of the past few bonded eras as well.

I don't know, I will check.

> Or is the idea that asking consensus is faster because it doesn't involve looking things up in the global state?

Yes, that as well.